API/Services/ImageService.cs

```diff
@@ -1,82 +1,164 @@
+    /// Checks if the specified file is an image.
     /// </summary>
-    /// <param name="filePath">Full path to the image to convert</param>
-    /// <param name="outputPath">Where to output the file</param>
-    /// <param name="encodeFormat">Encoding Format</param>
-    /// <returns>File of written encoded image</returns>
-    Task<string> ConvertToEncodingFormat(string filePath, string outputPath, EncodeFormat encodeFormat);
+    /// <param name="filePath">The path of the file to check.</param>
+    /// <returns>True if the file is an image; otherwise, false.</returns>
     Task<bool> IsImage(string filePath);
-    Task<string> DownloadFaviconAsync(string url, EncodeFormat encodeFormat);
-    Task<string> DownloadPublisherImageAsync(string publisherName, EncodeFormat encodeFormat);
+
+    /// <summary>
+    /// Downloads the favicon from the specified URL and saves it to the output directory.
+    /// </summary>
+    /// <param name="url">The URL of the favicon.</param>
+    /// <param name="encodeFormat">The encoding format to convert and save the favicon.</param>
+    /// <param name="quality">The quality of the favicon. Default is 100.</param>
+    /// <returns>The file name with extension of the saved favicon.</returns>
+    Task<string> DownloadFaviconAsync(string url, EncodeFormat encodeFormat, int quality = 100);
+
+    /// <summary>
+    /// Downloads the publisher image for the specified publisher name and saves it to the output directory.
+    /// </summary>
+    /// <param name="publisherName">The name of the publisher.</param>
+    /// <param name="encodeFormat">The encoding format to convert and save the publisher image.</param>
+    /// <param name="quality">The quality of the publisher image. Default is 100.</param>
+    /// <returns>The file name with extension of the saved publisher image.</returns>
+    Task<string> DownloadPublisherImageAsync(string publisherName, EncodeFormat encodeFormat, int quality = 100);
+
+    /// <summary>
+    /// Updates the color scape of an entity with a cover image.
+    /// </summary>
+    /// <param name="entity">The entity with a cover image.</param>
     void UpdateColorScape(IHasCoverImage entity);
+
+    /// <summary>
+    /// Replaces the file format of an image with the specified supported image formats by the browser, if needed, otherwise original file will be served.
+    /// </summary>
+    /// <param name="filename">The name of the image file.</param>
+    /// <param name="supportedImageFormats">The list of supported image formats by the browser.</param>
+    /// <param name="format">The encoding format to convert the image. Default is JPG</param>
+    /// <param name="quality">The quality of the image. Default is 99.</param>
+    /// <returns>The file name with extension of the replaced image file.</returns>
+    string ReplaceImageFileFormat(string filename, List<string> supportedImageFormats = null, EncodeFormat format = EncodeFormat.JPEG, int quality = 99);
```

Comment:
Hmm. The choice in here is JPEG, is because the encoding speed. PNG tends to be slower than JPG. Since, time in this case is essential to have a good Experience, I choose JPEG as a compromise. Anyway, this is from my own experience, not specifically for ImageMagick. Let met time log, both cases and we can compare numbers.

Reply:
Some data, backing the JPEG 99 choice:

Big JXL 48 Bit RGB Image (JXL Loading time 1.7 seconds)

Whole Conversion time (Loading & Saving)
```
WEBP - 100 -  7.3 seconds
AVIF - 100 -  4.2 seconds
PNG  - 100 - 11.1 seconds (Probably deflate max)
PNG  -   0 -  2.1 seconds (Most likely, uncompressed, big file)
JPEG -  99 -  1.9 seconds (Best bang for the buck)
JPEG - 100 -  3.4 seconds

```

So, JPEG 99, takes (1.9s - 1.7s) ~200 ms to save against the other formats.


Please NOTE:
I also added, Per Image Lock on conversion, to prevent multiple threads from processing the same image at the same time.